File: src/#226/solution.go
package solution

type TreeNode struct {
	Val         int
	Left, Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	if root != nil {
		invertTree(root.Right)
		invertTree(root.Left)
		if root.Left != nil || root.Right != nil {
			root.Right, root.Left = root.Left, root.Right
		}
		return root
	} else {
		return nil
	}
}

File: src/#226/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var invertTree = function (root) {
  if (root) {
    invertTree(root.left)
    invertTree(root.right)
    if (root.left || root.right) {  
        let temp=root.left
        root.left=root.right
        root.right=temp
    }
    return root
  } else {
    return null
  }
}
